OpenAPI/kiota/tutorials/dotnet-dependency-injection.md

```diff
@@ -0,0 +1,136 @@
+
+- HTTP ([Kiota default HttpClient-based implementation](https://github.com/microsoft/kiota-http-dotnet)) version `1.4.2` or higher
+- JSON serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-json-dotnet))
+- Form serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-form-dotnet))
+- Text serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-text-dotnet))
+- Multipart serialization ([Kiota default](https://github.com/microsoft/kiota-serialization-multipart-dotnet))
+- HttpClient support for Dependency Injection [Microsoft.Extensions.Http](https://www.nuget.org/packages/Microsoft.Extensions.Http)
+
+Run the following commands to get the required dependencies.
+
+```dotnetcli
+dotnet add package Microsoft.Extensions.Http
+dotnet add package Microsoft.Kiota.Abstractions
+dotnet add package Microsoft.Kiota.Http.HttpClientLibrary
+dotnet add package Microsoft.Kiota.Serialization.Json
+dotnet add package Microsoft.Kiota.Serialization.Form
+dotnet add package Microsoft.Kiota.Serialization.Text
+dotnet add package Microsoft.Kiota.Serialization.Multipart
+```
+
+## Generate the API client
+
+Kiota generates API clients from OpenAPI documents. Create a file named _github-releases.yml_ and add the following.
+
+:::code language="yaml" source="~/code-snippets/get-started/dotnet-dependency-injection/github-releases.yml":::
+
+You can then use the [Kiota](/openapi/kiota/install?tabs=bash&wt.mc_id=SEC-MVP-5004985#install-as-net-tool) command line tool to generate the API client classes.
+
+```bash
+kiota generate -l csharp -d github-releases.yml -c GitHubClient -n GitHub.ApiClient -o ./GitHub
+```
+
+## Create extension methods
+
+Create a new file named _KiotaServiceCollectionExtensions.cs_ in the root of your project and add the following code.
+
+:::code language="csharp" source="~/code-snippets/get-started/dotnet-dependency-injection/KiotaServiceCollectionExtensions.cs":::
+
+## Create a client factory
+
+Create a new file named _GitHubClientFactory.cs_ in the _GitHub_ folder and add the following code.
+
+This is what you call a factory pattern, where you create a factory class that is responsible for creating an instance of the client. This way you have a single place that is responsible for creating the client, and you can easily change the way the client is created. In this sample we will be registering the factory in the dependency injection container as well, because it will get the `HttpClient` injected in the constructor.
+
+:::code language="csharp" source="~/code-snippets/get-started/dotnet-dependency-injection/GitHub/GitHubClientFactory.cs":::
+
+## Register the API client
+
+Open the _Program.cs_ file and add the following code above the `var app = builder.Build();` line. Sample [Program.cs](https://github.com/microsoft/kiota-samples/blob/main/get-started/dotnet-dependency-injection/Program.cs).
+
+```csharp
```

Comment:
@baywet for clarity this code is still added to the docs, can I highlight specific lines in the included code block?

Reply:
I don't know about that, but maybe @jasonjoh can help. The question here is: when referencing a code file with the `:::code` syntax, is there a way to only reference specific lines? or highlight them?

Reply:
I will caution against using this heavily though - changes to the code between your snippet markers can cause the line numbers to change, which can throw off the highlighting if you don't remember to update that.